core/frontend/src/utils/pull_tracker.ts

```diff
@@ -0,0 +1,68 @@
+import { Dictionary } from '@/types/common'
+
+class PullTracker {
+    private layer_status: Dictionary<string> = {}
+
+    private layers: string[] = []
+
+    private layer_progress: Dictionary<string> = {}
+
+    private left_over_data = ''
+
+    pull_output = ''
+
+    private onready: () => any
```

Comment:
Why the return type is any and not void ?

Reply:
because I forgot what Ts used for void. oops.